src/resultSet.js

```diff
@@ -114,8 +114,7 @@ class ResultSet {
             let prop = props[i];
             var propIndex = prop[0];
             let prop_name = this._graph.getProperty(propIndex);
-            if (prop_name == undefined) {
-                console.error("fetching property labels");
+            while (prop_name == undefined) {
```

Comment:
Consider switching from while to a for loop going from 1 to X, after X tries, if we didn't managed to get the value, log and continue with an undefined value, apply this to all 3 fetches.